docs/src/analysis/sensitivity.md

```diff
@@ -1,36 +1,158 @@
+multiple dispatch then allows the internals of the solver to propagate the
+derivatives along the solution.
+
+#### Examples using ForwardDiff.jl
+
+The easiest way to use ForwardDiff.jl for local forward sensitivity analysis
+is to simply put the ODE solve inside of a function which you would like to
+differentiate. For example, let's define the ODE system for the Lotka-Volterra
+equations:
 
-#### Note
+```julia
+f = @ode_def begin
+  dx = a*x - b*x*y
+  dy = -c*y + d*x*y
+end a b c
+
+p = [1.5,1.0,3.0,1.0]
+u0 = [1.0;1.0]
+prob = ODEProblem(f,u0,(0.0,10.0),p)
+```
+
+Let's say we wanted to get the derivative of the final value w.r.t. each of
+the parameters. We can define the following function:
+
+```julia
+function test_f(p)
+  _prob = remake(prob;u0=convert.(eltype(p),prob.u0),p=p)
+  solve(prob,Vern9(),save_everystep=false)[end]
+end
+```
+
+What this function does is use the `remake` function
+[from the Problem Interface page](http://docs.juliadiffeq.org/latest/basics/problem.html#Modification-of-problem-types-1)
+to generate a new ODE problem with the new parameters, solves it, and returns
+the solution at the final time point. Notice that it takes care to make sure
+that the type of `u0` matches the type of `p`. This is because ForwardDiff.jl
+will want to use Dual numbers, and thus to propagate the Duals throughout
+the solver we need to make sure the initial condition is also of the type of
+Dual number. On this function we can call ForwardDiff.jl and it will return
+the derivatives we wish to calculate:
+
+```julia
+using ForwardDiff
+fd_res = ForwardDiff.jacobian(test_f,p)
+```
+
+If we would like to get the solution and the value at the time point at the
+same time, we can use [DiffResults.jl](https://github.com/JuliaDiff/DiffResults.jl).
+For example, the following uses a single ODE solution to calculate the value
+at the end point and its parameter Jacobian:
+
+```julia
+using DiffResults
+res = DiffResults.JacobianResult(u0,p) # Build the results object
+DiffResults.jacobian!(res,p) # Populate it with the results
+val = DiffResults.value(res) # This is the sol[end]
+jac = DiffResults.jacobian(res) # This is dsol/dp
+```
 
-Currently there are more performance optimizations needed to be done on the
-adjoint sensitivity method.
+If we would like to get the time series, we can do so by seeding the dual
+numbers directly. To do this, we use the `Dual` constructor. The first value
+is the value of the parameter. The second is a tuple of the derivatives. For
+each value we want to take the derivative by, we seed a derivative with a
+1 in a unique index. For example, we can build our parameter vector like:
 
-## Local Sensitivity Analysis
+```julia
+using ForwardDiff: Dual
+p1dual = Dual{Float64}(1.5, (1.0, 0.0, 0.0, 0.0))
+p2dual = Dual{Float64}(1.0, (0.0, 1.0, 0.0, 0.0))
+p3dual = Dual{Float64}(3.0, (0.0, 0.0, 1.0, 0.0))
+p4dual = Dual{Float64}(3.0, (0.0, 0.0, 0.0, 0.0))
+pdual = [p1dual, p2dual, p3dual, p4dual]
```

Comment:
@jrevels is there an easier way to seed Duals like this?

Reply:
This is actually really clean from a pedagogical standpoint. There is also [this constructor](https://github.com/JuliaDiff/ForwardDiff.jl/blob/master/src/dual.jl#L67), which allows you to write, for example:

```julia
using ForwardDiff: Dual, Chunk
chunk = Chunk{4}()
pdual = [Dual(1.5, chunk, Val(1)),
         Dual(1.0, chunk, Val(2)),
         Dual(3.0, chunk, Val(3)),
         Dual(3.0, chunk, Val(4))]
```

If you want to do it programmatically like ForwardDiff does it, see, for example, https://github.com/JuliaDiff/ForwardDiff.jl/blob/master/src/apiutils.jl#L17 and https://github.com/JuliaDiff/ForwardDiff.jl/blob/master/src/config.jl#L117.

BTW, the constructor is `Dual{T}(...)` where `T` is the tag type, not the element type (I'm assuming you don't actually want to use a `Float64` tag).

Reply:
I thought about writing out the programmatic way, but IMO that should be upstreamed to a convenience function in ForwardDiff. PR: https://github.com/JuliaDiff/ForwardDiff.jl/pull/379